.circleci/config.yml

```diff
@@ -0,0 +1,188 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  environment:
+    LEIN_ROOT: "true"
+    JVM_OPTS: -Xmx3200m
```

Comment:
Where you getting some errors without this?

Reply:
No. Just a carry over from another config. Can try without.

Reply:
Got it. I noticed this in the example provided by Circle themselves, so I guess that's how it became popular. I'm fine with it, I was just curious how this number was devised.

Reply:
It's important to set `JVM_OPTS` when running in CircleCI. Your jobs run on massive machines with tons of RAM, but inside a container (or similar) with a much smaller allocation of memory. By default older JVMs will set Xmx to a value higher than it is actually allowed to allocate. On quick builds, you may not generate enough garbage to hit the limit, or may only sometimes hit it. Newer JVM's (9 and 10 have both had improvements here) are smart enough to detect their effective max memory and set it by default, but JDK 8 doesn't have this IIRC.

Reply:
Good to know!